src/modules/utils.ts

```diff
@@ -112,6 +115,53 @@ export const runYarnIfPathExists = (relativePath: string) => {
   }
 }
 
+export const getPinnedDependencies = async (builderHttp: AxiosInstance) => {
+  try {
+    const result = await builderHttp.get('/_v/builder/0/pinneddeps')
+    return result.data
+  } catch (e) {
+    log.debug('Failed to connect to builder-hub to get pinned deps')
+    throw e
+  }
+}
+
+const getEntryMapFromObject = (obj: object, field: string) => {
+  if (obj.hasOwnProperty(field)) {
+    return new Map<string, string>(Object.entries(obj[field]))
+  }
+  else {
+    return new Map<string, string>()
+  }
+}
+
+// For each entry in 'left', get all keys that exist but differ in 'right'
+const leftMapDifference = (left: Map<string, string>, right: Map<string, string>) => {
+  return R.filter(x => right.has(x) && left.get(x) !== right.get(x), [...left.keys()])
+}
+
+export const fixPinnedDependencies = R.curry(async (pinnedDeps: Map<string, string>, relativePath: string) => {
+  const jsonPath = resolvePath(getAppRoot(), `${relativePath}/package.json`)
+  if (!await pathExists(jsonPath)) {
+    return
+  }
+  const packageJSON = JSON.parse((await readFile(jsonPath)).toString())
+  const dependencies = getEntryMapFromObject(packageJSON, 'dependencies')
+  const devDependencies = getEntryMapFromObject(packageJSON, 'devDependencies')
+  const outdatedDeps = leftMapDifference(dependencies, pinnedDeps)
+  const outdatedDevDeps = leftMapDifference(devDependencies, pinnedDeps)
+  const newPackageJSON = R.reduce((obj, dep) => {
+    log.warn(`${dep} is outdated. Upgrading to ${pinnedDeps.get(dep)}`)
```

Comment:
I think we should change this warning because the user's dependencies might not necessarily be outdated.